### ISSUE TITLE
detection_count =0 during training 

### ISSUE THREAD

**Issue by shriprateek** [AlexeyAB/darknet#1584]
While training YOLO_V3 for custom dataset, till 4200 iterations, my training set mAP went upto 37%. But after that in the 4300th iteration, the training set mAP went straight to 0 % because of 0 detections among 21542 detections. What should I interpret from this and how to resolve this issue ?

**Comment by AlexeyAB** [comment]
Look at the files `bad.list` and `bad_label.list` if the exist.
Look are there warinings during training.
And check you dataset by using: https://github.com/AlexeyAB/Yolo_mark

**Comment by AlexeyAB** [comment]
Also attach your cfg-file.
And what params do you use in the Makefile?

**Comment by shriprateek** [comment]
Thanks AlexeyAB, When I opened my images with yolo_mark, I found that the bounding box was not fitting the object correctly. I streched from centre of the object to one corner. I am retraining with the proper data on yoloV3.
Could you please explain what the below terms mean:
1)detections_count
2)unique_truth_count.

And how can I see which images were analysed as FP and FN to understand what kind of training data needs to be added.

**Comment by AlexeyAB** [comment]
* detections_count - total detections by Yolo for all images with `probability > 0.005`
* unique_truth_count - total bounded boxes from your validation dataset

> And how can I see which images were analysed as FP and FN to understand what kind of training data needs to be added.

It isn't implemented.
You should change source code in this place and may be other places: https://github.com/AlexeyAB/darknet/blob/ca43bbdaaede5c9cbf82a8a0aa5e2d0a4bdcabc0/src/detector.c#L734-L740

**Comment by shriprateek** [comment]
After correcting the bounding boxes, still in the bad.list I am finding alot of my training set images and label text files. I am attaching my cfg file with this comment . Please let me know what are these bad labels and what can I do to fix this. To answer your earlier question, no there are no warnings during training.

[tad_yolov3.txt](https://github.com/AlexeyAB/darknet/files/2378882/tad_yolov3.txt)

**Comment by AlexeyAB** [comment]
Your Makefile and tad_yolov3.cfg looks good.

* Can you attach your bad.list and bad_label.list?
* Also try to update your code from this GitHub to the latest commit.

**Comment by AlexeyAB** [comment]
So you use incorrect paths to these images.
Or remove this new image files from `train.txt` or move your images and txt-labels to the correct path.

Also try to update your code from GitHub, and remove old bad.list and bad_label.list files.
